Library/Homebrew/cmd/update.sh

```diff
@@ -726,8 +726,11 @@ EOS
         if ! git fetch --tags --force "${QUIET_ARGS[@]}" origin \
            "refs/heads/${UPSTREAM_BRANCH_DIR}:refs/remotes/origin/${UPSTREAM_BRANCH_DIR}" 2>>"${tmp_failure_file}"
         then
+          local fetch_error=
+          2>/dev/null read -rd '' fetch_error <"${tmp_failure_file}" || :
+
           # Reprint fetch errors to stderr
-          [[ -f "${tmp_failure_file}" ]] && cat "${tmp_failure_file}" 1>&2
+          echo -ne "${fetch_error:+${fetch_error}\\\n}" 1>&2
```

Comment:
```suggestion
          [[ -n "${fetch_error}" ]] && echo "${fetch_error}" 1>&2
```
If this is equivalent, it seems much easier to read.

Reply:
Result-wise, what you suggest is almost equivalent. The main difference is the exit code: if `fetch_error` is empty, your suggestion returns 1. To avoid non-zero exit code, we'd have to write `if [[ -n $x ]]; then echo $x; fi`

Reply:
Why do we need to avoid the non-zero exit code? Does it make `brew update` exit with an error?

Reply:
Why wouldn't we use `${x:+$x}`? This is a standard parameter expansion in Bash, is just as readable, fast, and doesn't return a non-zero exit code.

Reply:
> is just as readable

It's nowhere near readable for anyone who doesn't know Bash extremely well (which is not the vast majority of contributors or maintainers to this project).